buildSrc/subprojects/build/src/main/groovy/org/gradle/build/docs/AssertNoDaemonAsciidoctorTask.groovy

```diff
@@ -0,0 +1,38 @@
+/*
+ * Copyright 2018 the original author or authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.gradle.build.docs
+
+import org.gradle.internal.environment.GradleBuildEnvironment
+import javax.inject.Inject
+import org.asciidoctor.gradle.AsciidoctorTask
+
+class AssertNoDaemonAsciidoctorTask extends AsciidoctorTask {
+    private GradleBuildEnvironment buildEnvironment
+
+    @Inject
+    AssertNoDaemonAsciidoctorTask(GradleBuildEnvironment buildEnvironment) {
+        this.buildEnvironment = buildEnvironment
+    }
+
+    @Override
+    void processAsciidocSources() {
+        // https://github.com/gradle/gradle-private/issues/1352
+        // JRuby has a serious issue that sometimes its native library makes the whole JVM crash
+        // We have to disable daemon for this task as a workaround
+        assert !buildEnvironment.longLivingProcess
```

Comment:
I guess we should only check this when on `CI`. Developers can run the Asciidoctor tasks in the daemon, right?